source/fundamentals/logging.txt

```diff
@@ -0,0 +1,285 @@
+         </dependency>
+
+      Once you have installed the above dependencies, connect to your
+      MongoDB instance and retrieve a document with the following code.
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create(<my uri>);
+         MongoDatabase database = mongoClient.getDatabase(<my database>);
+         MongoCollection<Document> collection = database.getCollection(<my collection>);
+         collection.find().first();
+
+      You should see output resembling the following:
+
+      .. code-block:: none
+
+         ...
+         1156 [main] DEBUG o.m.driver.protocol.command - Sending command '{"find": "<your collection>", "filter": {}, "limit": 1, "singleBatch": true, "$db": "<your db>", "lsid": {"id": {"$binary": {"base64": "<server session>", "subType": "04"}}}}' with request id 13 to database <your db> on connection <your connection> to server <your server>
+         1173 [main] DEBUG o.m.driver.protocol.command - Execution of command with request id 13 completed successfully in 30.01 ms on connection <your connection> to server <your server>
+   
+      For more information on Logback, see the
+      `Logback manual <http://logback.qos.ch/manual/>`__.
+   
+   .. tab:: Log4j2
+      :tabid: Log4j2-binding
+   
+      If you add these dependencies to your ``pom.xml`` file, the MongoDB Java
+      driver uses the Log4j2 framework to output your logs.
+
+      .. code-block:: xml      
+      
+         <!-- SLF4J API -->
+         <dependency>
+           <groupId>org.slf4j</groupId>
+           <artifactId>slf4j-api</artifactId>
+           <version>1.7.30</version>
+         </dependency>
+         <!-- SLF4J-Log4j2 binding -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-slf4j-impl</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+         <!-- Log4j2 core functionality -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-core</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+
+      Once you have installed the above dependencies, log an error using the
+      following code.
+
+      .. code-block:: java
+
+         import org.slf4j.Logger;
+         import org.slf4j.LoggerFactory;
+
+         ...
+
+         Logger logger = LoggerFactory.getLogger(App.class);
+         logger.error("Logging an Error");
+
+      You should see output resembling the following:
+
+      .. code-block:: none
+
+         12:35:00.438 [main] ERROR <my package path> - Logging an Error
+
+      For more information on Log4j2, see the
+      `Log4j2 manual <https://logging.apache.org/log4j/2.x/manual/>`__.
+
+Configure Your Logger
+---------------------
+
+To configure your logger, you must configure the logging framework bound to
+SLF4J.
+
+The following subsection provides a concrete example showing how to configure
+your logger.
```

Comment:
Do these provide any meaning or instruction more than the headings already do?

Reply:
Thanks for the note. I believe the first sentence does (it gives explicit guidance for going about configuring your logger that I do not think fits naturally into the example), but the second sentence does not. I have removed the second sentence and changed the heading of the subsection to "Example". Fix: